src/handler/description.rs

```diff
@@ -0,0 +1,257 @@
+    fn entry() -> Self;
+
+    /// Description for a user-defined handler that can do practically
+    /// everything.
+    fn user_defined() -> Self;
+
+    /// Merge descriptions to get a description for a chain handler.
+    fn merge_chain(&self, other: &Self) -> Self;
+
+    /// Merge descriptions to get a description for a branch handler.
+    fn merge_branch(&self, other: &Self) -> Self;
+
+    /// Description for [`map`](crate::map).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn map() -> Self {
+        Self::user_defined()
+    }
+
+    /// Description for [`map_async`](crate::map_async).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn map_async() -> Self {
+        Self::user_defined()
+    }
+
+    /// Description for [`filter`](crate::filter).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn filter() -> Self {
+        Self::user_defined()
+    }
+
+    /// Description for [`filter_async`](crate::filter_async).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn filter_async() -> Self {
+        Self::user_defined()
+    }
+
+    /// Description for [`filter_map`](crate::filter_map).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn filter_map() -> Self {
+        Self::user_defined()
+    }
+
+    /// Description for [`filter_map_async`](crate::filter_map_async).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn filter_map_async() -> Self {
+        Self::user_defined()
+    }
+
+    /// Description for [`endpoint`](crate::endpoint).
+    ///
+    /// ## Default implementation
+    ///
+    /// By default this returns the value from
+    /// [`user_defined`](HandlerDescription::user_defined).
+    #[track_caller]
+    fn endpoint() -> Self {
+        Self::user_defined()
+    }
+}
+
+/// Uninformative handler description.
+#[derive(Debug, PartialEq, Eq)]
+pub struct Unspecified(());
+
+/// Description for a handler that describes what event kinds are interesting to
+/// the handler.
+#[derive(Debug, Clone)]
+pub enum EventKindDescription<K, S = RandomState> {
+    /// Only event) kinds in the set are "interesting".
+    InterestingEventKinds(HashSet<K, S>),
```

Comment:
"Interesting" sounds a bit peculiar to me. But maybe it's just me. It seems that `epoll` uses the term "interest list", which is almost the same as we use here.

Reply:
I'm not a fun of "interesting" either, but I don't know how to better describe "possibly observable by the endpoints or user defined handlers (`from_fn`)" 😅